examples/bond/scripts/run/bond-functions.js

```diff
@@ -0,0 +1,328 @@
+  const optInTx = [
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: storeManagerAccount,
+      toAccountAddr: issuerLsig.address(),
+      amountMicroAlgos: 0,
+      payFlags: {}
+    },
+    {
+      type: types.TransactionType.OptInASA,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: issuerLsig.address(),
+      lsig: issuerLsig,
+      assetID: newIndex,
+      payFlags: {}
+    }
+  ];
+  await executeTransaction(deployer, optInTx);
+
+  scInitParam = {
+    TMPL_OLD_BOND: asaInfo.assetIndex,
+    TMPL_NEW_BOND: newIndex,
+    TMPL_APPLICATION_ID: appInfo.appID,
+    TMPL_STORE_MANAGER: storeManagerAccount.addr
+  };
+  const dexLsig = await deployer.loadLogic('dex-lsig.py', scInitParam);
+
+  await fundAccount(deployer, dexLsig.address());
+
+  optInTx[0].toAccountAddr = dexLsig.address();
+  optInTx[1].fromAccountAddr = dexLsig.address();
+  optInTx[1].lsig = dexLsig;
+  await executeTransaction(deployer, optInTx);
+
+  optInTx[1].assetID = asaInfo.assetIndex;
+  await executeTransaction(deployer, optInTx);
+
+  const globalState = await readGlobalStateSSC(deployer, storeManagerAccount.addr, appInfo.appID);
+  let total = 0;
+  for (const l of globalState) {
+    const key = Buffer.from(l.key, 'base64').toString();
+    if (key === 'total') {
+      total = l.value.uint;
+      break;
+    }
+  }
+
+  // Transfer total amount to dex lsig
+  const transferTx = {
+    type: types.TransactionType.TransferAsset,
+    sign: types.SignType.SecretKey,
+    fromAccount: creatorAccount,
+    toAccountAddr: dexLsig.address(),
+    amount: total,
+    assetID: newIndex,
+    payFlags: { totalFee: 1000 }
+  };
+  await executeTransaction(deployer, transferTx);
+
+  // balance of old bond tokens in issuer lsig
+  const info = await balanceOf(deployer, issuerLsig.address(), asaInfo.assetIndex);
+  console.log('Old balance amount ', info.amount);
+  const groupTx = [
+    // call to bond-dapp
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: storeManagerAccount,
+      appID: appInfo.appID,
+      payFlags: {},
+      appArgs: ['str:create_dex'],
+      accounts: [issuerLsig.address()]
+    },
+    // New bond token transfer to issuer's address
+    {
+      type: types.TransactionType.TransferAsset,
+      sign: types.SignType.SecretKey,
+      fromAccount: creatorAccount,
+      toAccountAddr: issuerLsig.address(),
+      amount: info.amount,
+      assetID: newIndex,
+      payFlags: { totalFee: 1000 }
+    },
+    // burn tokens
+    {
+      type: types.TransactionType.TransferAsset,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: issuerLsig.address(),
+      lsig: issuerLsig,
+      toAccountAddr: creatorAccount.addr,
+      amount: info.amount,
+      assetID: asaInfo.assetIndex,
+      payFlags: { totalFee: 1000 }
+    }
+  ];
+
+  console.log('Creating dex!');
```

Comment:
```suggestion
  console.log('Creating dex %d!', i);
```

Reply:
`i` is the new parameter to this function

Reply:
done